subsys/net/ip/net_if.c

```diff
@@ -4159,6 +4159,20 @@ static void update_operational_state(struct net_if *iface)
 	}
 }
 
+static void init_igmp(struct net_if *iface)
+{
+#if defined(CONFIG_NET_IPV4_IGMP)
+	/* Ensure IPv4 is enabled for this interface */
+	if (iface->config.ip.ipv4 == NULL) {
+		return;
+	}
+
+	net_ipv4_igmp_init(iface);
+#else
+	return;
```

Comment:
Minor nit but just in case we could add here `ARG_UNUSED(iface);` to the else branch.

Reply:
Thanks, I've updated the code.